### DISCUSSION TITLE
Use_keys without a key specified

### DISCUSSION THREAD

**Discussion by blademd** [ktbyers/netmiko#3333] [Ideas]
Hello!

What if add a flag to the BaseConnection constructor to ignore the `self._key_check()` call if `use_keys` is set and this new flag is also set? The idea behind that is to allow Paramiko to search for any possible key instead of specifying one manually.

**Comment by ktbyers** [comment]
Under this model, what does Paramiko report as an error when it can't find a usable key?

**Reply by blademd** [reply]
It depends. If there are no keys, Paramiko throws an exception "paramiko.ssh_exception.SSHException: No authentication methods available." If there are some keys, but a host does not recognize them, there is an exception "paramiko.ssh_exception.AuthenticationException: Authentication failed."
Let's say, if we have a `password` specified instead of a `passphrase` to decrypt a key, Paramiko tries it in the end for a password-based authentication. In case of failure, it also throws "paramiko.ssh_exception.AuthenticationException: Authentication failed."
